app/lib/methods/getRoles.js

```diff
@@ -2,9 +2,58 @@ import { sanitizedRaw } from '@nozbe/watermelondb/RawRecord';
 
 import database from '../database';
 import log from '../../utils/log';
+import reduxStore from '../createStore';
 import protectedFunction from './helpers/protectedFunction';
+import {
+	addRoles, removeRoles, setRoles as setRolesAction, updateRoles
+} from '../../actions/roles';
 
-export default function() {
+export async function setRoles() {
+	const db = database.active;
+	const rolesCollection = db.collections.get('roles');
+	const allRoles = await rolesCollection.query().fetch();
+	const parsed = allRoles.reduce((acc, item) => ({ ...acc, [item.id]: item.description || item.id }), {});
+	reduxStore.dispatch(setRolesAction(parsed));
+}
+
+export async function subscribeRoles(ddpMessage) {
```

Comment:
This function is receiving data from DDP, updating local db and dispatching new values to Redux store.
We should rename it to something like `onRolesChanged`